### PR TITLE
Added a format method for OpenCL expressions in elementwise operations

### PR DESCRIPTION
**Pull request by lambday**
@karlnapf looks way much cleaner now, a la @lisitsyn.

### CODE REVIEW

**Comment by lisitsyn** [comment]
Lookz cool

**Comment by lambday** [comment]
@lisitsyn merging it then.
@karlnapf Please have a look when you're free :)

File: benchmarks/elementwise_benchmark.cpp [hunk 1]
```diff
@@ -115,16 +115,24 @@ BENCHMARK(CGPUMatrix, elementwise, 10, 1000)
...
+	{
+		linalg::ocl::Parameter("type")=data_type,
+		linalg::ocl::Parameter("var1")="outer_factor",
+		linalg::ocl::Parameter("var2")="exp_factor",
+		linalg::ocl::Parameter("var3")="sin_factor",
+		linalg::ocl::Parameter("pi")=CMath::PI,
+		linalg::ocl::Parameter("weights")=weights,
+		linalg::ocl::Parameter("stddev")=std_dev,
+		linalg::ocl::Parameter("mean")=mean
+	}));
```
Comment (karlnapf) [inline benchmarks/elementwise_benchmark.cpp]: much cleaner   +1

File: tests/unit/mathematics/linalg/ElementwiseOperations_unittest.cc [hunk 2]
```diff
@@ -354,22 +362,30 @@ TEST(Elementwise_custom, CGPUMatrix_inplace)
...
+		linalg::ocl::Parameter("pi")=CMath::PI,
+		linalg::ocl::Parameter("weights")=weights,
+		linalg::ocl::Parameter("stddev")=std_dev,
+		linalg::ocl::Parameter("mean")=mean
+	}));
 
 	linalg::elementwise_compute_inplace(m_gpu, operation);
 
-	for (index_t i=0; i<m.num_rows; ++i)
+	for (auto i=0; i<m.num_rows; ++i)
```
Comment (karlnapf) [inline tests/unit/mathematics/linalg/ElementwiseOperations_unittest.cc]: ah nice! finally

**Comment by karlnapf** [comment]
Good stuff, thanks!

**Pull request merged**